### PR TITLE
boards/saml21-xpro: add Arduino API support for saml21-xpro board

### PR DESCRIPTION
**Pull request by fedepell**
### Contribution description
Adds the Arduino API support for the Atmel SAML21-XPRO board.
This enables Arduino-like programming supported by RIOT also on this board.

### Testing procedure
Arduino automatic test sys_arduino can be used for basic functionalities tests:

```
cd tests/sys_arduino/
BOARD=saml21-xpro make all flash test
```

Additionally one can manually use and play with the examples/arduino_hello-world/ for other tests (ie. playing with I/O and leds)

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
I'm not too familiar with the Arduino wrappers but don't we usually provide a "arduino_pinmap.h" file to create pin Arduino<->RIOT mappings? @aabadie?

**Comment by fedepell** [comment]
@PeterKietzmann : thanks for the feedback! Indeed I checked more examples and saw that most of them have the files split into two .h as you suggest. So I modified my PR splitting into two and squashed together.
Hope it looks better and more in line with other boards now!

**Comment by dylad** [comment]
Changes look good but I'm wondering what are the benefits from adding arduino support for SAML21 ? Is there any arduino board featuring this MCU ? or using the same header connectors ?

**Comment by fedepell** [comment]
In my case (don't know if this can be generalized) the main point was to enable some internal developers (more hw oriented) with Arduino knowledge and already existing code to develop on this board.

**Comment by fedepell** [comment]
@dylad : it wasn't the scope of my PR but now I noticed that indeed there is a SAM arduino-like board indeed out there:
https://startingelectronics.org/articles/arduino/wemos-arduino-m0/
So maybe it would be more interesting in this aspect. I will try to get my hands on the board and add support for it too.

**Comment by dylad** [comment]
@fedepell the board you link here is samd21-based so this won't help with this PR.
The only saml21-based board that I know is saml21-xpro from Atmel but there is no arduino connectors on it so I am really puzzle with this PR.

**Comment by fedepell** [comment]
@dylad: no worries about this PR! If you think my use case could not be useful to others and the project no worries just close it, I just made the PR thinking it may be of use to someone else as well :) I can manage this small thing in our local repository!

**Comment by PeterKietzmann** [comment]
@dylad we don't only support the arduino API for boards with arduino pinout. For example, we have esp32 or stm32disco boards implementing it.

**Comment by PeterKietzmann** [comment]
I did a basic code review and set the related labels. Unfortunately I'm not able to provide any testing here

**Comment by dylad** [comment]
@PeterKietzmann Good to know ! I'll handle tests this week so this PR can finally be merged.

File: boards/saml21-xpro/include/arduino_pinmap.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * Copyright (C)  2018 Federico Pellegrin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_saml21-xpro
+ * @brief       Mappinf from board pins to Arduino pins
```
Comment (dylad) [inline boards/saml21-xpro/include/arduino_pinmap.h]: Mapping

**Comment by dylad** [comment]
Please fix typo before we merge :) you can squash directly.

File: boards/saml21-xpro/include/arduino_pinmap.h (hunk 1 above)
Comment (fedepell) [reply boards/saml21-xpro/include/arduino_pinmap.h]: fixed now, thanks!

**Comment by fedepell** [comment]
Thanks for checking! Just corrected the typo and squashed!

**Comment by dylad** [comment]
All green -> Let's go.

**Pull request merged**